src/audio/codec_adapter/codec_adapter.c

```diff
@@ -406,6 +406,13 @@ static int codec_adapter_copy(struct comp_dev *dev)
 			break;
 		}
 
+		/* TODO: remove magic number and use possible maximum output per
+		 * decoding algorithm type
+		 * 9216 is maximum frame length bytes for mp3 at 192Khz, 24bits,
```

Comment:
that makes no sense. mp3 never supported 192kHz and perceptual coders don't care about bit depth.

I don't recall what the max value was but this was something at 320 kbits/s

Reply:
@plbossart It is clear that I still need to read upon mp3 stuff and all. I'm still confused about all the version / etc. 

I started with a sample mp3 file:

```
root@imx8qxpc0mek:~# file samples/lb.mp3    
samples/lb.mp3: MPEG ADTS, layer III, v1, 192 kbps, 48 kHz, Stereo
```

From the internet I have this:

```
// Bitrates - use [version][layer][bitrate]
const uint16_t mpeg_bitrates[4][4][16] = {
  { // Version 2.5
    { 0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0, 0 }, // Reserved
    { 0,   8,  16,  24,  32,  40,  48,  56,  64,  80,  96, 112, 128, 144, 160, 0 }, // Layer 3
    { 0,   8,  16,  24,  32,  40,  48,  56,  64,  80,  96, 112, 128, 144, 160, 0 }, // Layer 2
    { 0,  32,  48,  56,  64,  80,  96, 112, 128, 144, 160, 176, 192, 224, 256, 0 }  // Layer 1
  },
  { // Reserved
    { 0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0, 0 }, // Invalid
    { 0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0, 0 }, // Invalid
    { 0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0, 0 }, // Invalid
    { 0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0, 0 }  // Invalid
  },
  { // Version 2
    { 0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0, 0 }, // Reserved
    { 0,   8,  16,  24,  32,  40,  48,  56,  64,  80,  96, 112, 128, 144, 160, 0 }, // Layer 3
    { 0,   8,  16,  24,  32,  40,  48,  56,  64,  80,  96, 112, 128, 144, 160, 0 }, // Layer 2
    { 0,  32,  48,  56,  64,  80,  96, 112, 128, 144, 160, 176, 192, 224, 256, 0 }  // Layer 1
  },
  { // Version 1
    { 0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0, 0 }, // Reserved
    { 0,  32,  40,  48,  56,  64,  80,  96, 112, 128, 160, 192, 224, 256, 320, 0 }, // Layer 3
    { 0,  32,  48,  56,  64,  80,  96, 112, 128, 160, 192, 224, 256, 320, 384, 0 }, // Layer 2
    { 0,  32,  64,  96, 128, 160, 192, 224, 256, 288, 320, 352, 384, 416, 448, 0 }, // Layer 1
  }
};
```

If my understanding is correct MP3 is  MPEG-1 Audio Layer III or MPEG-2 Audio Layer III.

So, based on the table above mp3 should also support 192kHZ. But I admit my comment there is just bad! I will remove it.

Reply:
you may be confusing kbs and kHz :-)

the rates are 
version 1: 48, 44.1 and 32kHz
version 2: 24, 22.05 and 16 kHz
version 2.5: 12, 11.025 and 8 kHz (this is actually not a standard but an extension by FhG).

https://en.wikipedia.org/wiki/MP3 has most of the needed information.

Reply:
Yes, you are right the comment should be read 192kbps.